mediapipe/tasks/python/vision/gesture_recognizer.py

```diff
@@ -0,0 +1,467 @@
+        ], [
+          [landmark_module.NormalizedLandmark.create_from_pb2(hand_landmark)
+           for hand_landmark in hand_landmarks.landmark]
+          for hand_landmarks in hand_landmarks_proto_list
+        ], [
+          [landmark_module.Landmark.create_from_pb2(hand_world_landmark)
+           for hand_world_landmark in hand_world_landmarks.landmark]
+          for hand_world_landmarks in hand_world_landmarks_proto_list
+        ]
+      )
+      timestamp = output_packets[_HAND_GESTURE_STREAM_NAME].timestamp
+      options.result_callback(
+          gesture_recognition_result, image,
+          timestamp.value // _MICRO_SECONDS_PER_MILLISECOND)
+
+    task_info = _TaskInfo(
+        task_graph=_TASK_GRAPH_NAME,
+        input_streams=[
+            ':'.join([_IMAGE_TAG, _IMAGE_IN_STREAM_NAME]),
+            ':'.join([_NORM_RECT_TAG, _NORM_RECT_STREAM_NAME]),
+        ],
+        output_streams=[
+          ':'.join([_HAND_GESTURE_TAG, _HAND_GESTURE_STREAM_NAME]),
+          ':'.join([_HANDEDNESS_TAG, _HANDEDNESS_STREAM_NAME]),
+          ':'.join([_HAND_LANDMARKS_TAG, _HAND_LANDMARKS_STREAM_NAME]),
+          ':'.join([_HAND_WORLD_LANDMARKS_TAG,
+                    _HAND_WORLD_LANDMARKS_STREAM_NAME]),
+          ':'.join([_IMAGE_TAG, _IMAGE_OUT_STREAM_NAME])
+        ],
+        task_options=options)
+    return cls(
+        task_info.generate_graph_config(
+            enable_flow_limiting=options.running_mode ==
+            _RunningMode.LIVE_STREAM), options.running_mode,
+        packets_callback if options.result_callback else None)
+
+  def recognize(
+      self,
+      image: image_module.Image,
+      image_processing_options: Optional[_ImageProcessingOptions] = None
+  ) -> GestureRecognitionResult:
+    """Performs hand gesture recognition on the given image. Only use this
+    method when the GestureRecognizer is created with the image running mode.
+
+    The image can be of any size with format RGB or RGBA.
+    TODO: Describes how the input image will be preprocessed after the yuv
+    support is implemented.
+
+    Args:
+      image: MediaPipe Image.
+      image_processing_options: Options for image processing.
+
+    Returns:
+      The hand gesture recognition results.
+
+    Raises:
+      ValueError: If any of the input arguments is invalid.
+      RuntimeError: If gesture recognition failed to run.
+    """
+    normalized_rect = self.convert_to_normalized_rect(image_processing_options,
+                                                      roi_allowed=False)
+    output_packets = self._process_image_data({
+        _IMAGE_IN_STREAM_NAME: packet_creator.create_image(image),
+        _NORM_RECT_STREAM_NAME: packet_creator.create_proto(
+            normalized_rect.to_pb2())})
+
+    if output_packets[_HAND_GESTURE_STREAM_NAME].is_empty():
+      return GestureRecognitionResult([], [], [], [])
+
+    gestures_proto_list = packet_getter.get_proto_list(
```

Comment:
L316 -L349 can be extracted to be a separate method that can be shared by `recognize`, `recognize_for_video`, and `packet_callback`.

Reply:
@jiuqiant I've refactored the code so that those methods reuse the separate method.